lib/tapioca/commands/annotations.rb

```diff
@@ -74,22 +83,33 @@ def fetch_index
       sig { params(gem_names: T::Array[String]).returns(T::Array[String]) }
       def fetch_annotations(gem_names)
         say("Fetching gem annotations from central repository... ", [:blue, :bold])
-        fetchable_gems = gem_names.select { |gem_name| @index.has_gem?(gem_name) }
+        fetchable_gems = T.let(gem_names.each_with_object({}) do |gem_name, hash|
+          @indexes.each do |uri, index|
+            if index.has_gem?(gem_name)
+              hash[gem_name] ||= []
+              hash[gem_name] << uri
+            end
+          end
+        end, T::Hash[String, T::Array[String]])
 
         if fetchable_gems.empty?
           say(" Nothing to do")
           exit(0)
         end
 
         say("\n")
-        fetched_gems = fetchable_gems.select { |name| fetch_annotation(name) }
+        fetched_gems = fetchable_gems.select { |gem_name, repo_uris| fetch_annotation(repo_uris, gem_name) }
```

Comment:
Nitpick: the opposite order of block args and `fetch_annotation` params feels odd. Is it possible to reverse the order of `fetch_annotation` params? I think `fetch_annotation(gem_name, repo_uris)` also makes sense and reads better here.

Reply:
I kept the same order than the other methods (`fetch_index`, `fetch_local_file`, etc.) where the repo URI comes first.